plugins/hypervisors/xenserver/src/com/cloud/hypervisor/xenserver/resource/CitrixResourceBase.java

```diff
@@ -2645,7 +2649,7 @@ private String actualIsoTemplate(final Connection conn) throws BadServerResponse
         final String[] items = xenVersion.split("\\.");
 
         // guest-tools.iso for XenServer version 7.0+
-        if (xenBrand.equals("XenServer") && Integer.parseInt(items[0]) >= 7) {
+        if ((xenBrand.equals("XenServer") || xenBrand.equals("XCP-ng")) && Integer.parseInt(items[0]) >= 7) {
```

Comment:
just a comment; how about defining applicable targets as a set for readability.
Also, though it doesn't apply to this PR, parsing an it is not a very prozaic way of saying you need to be at version 7 or higher. that comment at 2651 should have been a method name instead of a comment

Reply:
Not sure how to address this, I'll let it be for this PR.